quarkus-workshop-super-heroes/docs/src/docs/asciidoc/5a-contract-testing/states.adoc

```diff
@@ -0,0 +1,55 @@
+== States
+
+One of the advantages of Pact over simpler contract verification methods, such as syntax-checking, is that a pact can include a range of expected responses, depending on what was passed in, and external state.
+
+This allows for a richer set of checks.
+For example, it can exercise more interesting logic on the consumer side, and it can act as a proper functional test for the provider.
+
+Let's exercise some error-handling.
+What happens if no heroes can be found?
+In that case, we expect that the heroes server will return a 404. Let's get that into the contract, and exercise the fight service's error handling.
+
+Add a new pact.
+We use the `given()` to identify the state.
+[source,java]
+
+----
+@Pact(consumer = "rest-fights")
+    public V4Pact randomHeroNotFoundPact(PactDslWithProvider builder) {
+        return builder
+            .given("No random hero found")
+            .uponReceiving("A request for a random hero")
+            .path(HERO_RANDOM_URI)
+            .method(HttpMethod.GET)
+            .headers(HttpHeaders.ACCEPT, MediaType.APPLICATION_JSON)
+            .willRespondWith()
+            .status(Status.NOT_FOUND.getStatusCode())
+            .toPact(V4Pact.class);
+    }
+----
```

Comment:
Maybe use some indentation to make it easier to read (like the first contract)?

Reply:
The indentation is a bit off (the `@Pact` annotation doesn't line up with the method. Not sure if its a big deal or not?